ocaml/sdk-gen/java/main.ml

```diff
@@ -1052,6 +1061,29 @@ public class Types
   TypeSet.iter (gen_task_result_func file) !types ;
   fprintf file
     {|
+    
+    public static class BadAsyncResult extends XenAPIException
+    {
+        public final String result;
+
+        public BadAsyncResult(String result)
+        {
+            super(result);
+            this.result = result;
+        }
+    }
+
+    private static String parseResult(String result) throws BadAsyncResult
```

Comment:
Should the methods `toX()`, which call `parseResult()`, also be marked with `throws BadAsyncResult`?

Reply:
Hmmm, on second thought probably not - if it happens it's our problem, the user should not be expected to handle this as a matter of course.